File: frontend/src/types/template.ts
export type TemplateType = 'task' | 'project' | 'triage';

export interface WorkTemplate {
    id: number;
    name: string;
    description?: string | null;
    seed_key?: string | null;
    template_type: TemplateType;
    default_title?: string | null;
    default_description?: string | null;
    default_priority?: number | null;
    default_effort_days?: number | null;
    default_labels: string[];
    default_checklist: string[];
    default_payload: Record<string, unknown>;
    is_active: boolean;
    sort_order: number;
    created_at: string;
    updated_at: string;
}

export interface WorkTemplateCreate {
    name: string;
    description?: string | null;
    template_type: TemplateType;
    default_title?: string | null;
    default_description?: string | null;
    default_priority?: number | null;
    default_effort_days?: number | null;
    default_labels?: string[];
    default_checklist?: string[];
    default_payload?: Record<string, unknown>;
    is_active?: boolean;
    sort_order?: number;
}

export type WorkTemplateUpdate = Partial<WorkTemplateCreate>;

export interface TemplateListParams {
    template_type?: TemplateType;
    include_inactive?: boolean;
}

File: backend/app/schemas/template.py
"""Reusable work template schemas."""
from datetime import datetime
from enum import Enum
from typing import Any, Optional

from pydantic import BaseModel, ConfigDict, Field


class TemplateType(str, Enum):
    """Template target type."""
    TASK = "task"
    PROJECT = "project"
    TRIAGE = "triage"


class WorkTemplateCreate(BaseModel):
    """Schema for creating a reusable work template."""
    model_config = ConfigDict(extra="forbid")

    name: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    template_type: TemplateType
    default_title: Optional[str] = Field(default=None, min_length=1, max_length=500)
    default_description: Optional[str] = None
    default_priority: Optional[int] = Field(default=None, ge=1, le=10)
    default_effort_days: Optional[float] = Field(default=None, ge=0.1)
    default_labels: list[str] = Field(default_factory=list)
    default_checklist: list[str] = Field(default_factory=list)
    default_payload: dict[str, Any] = Field(default_factory=dict)
    is_active: bool = True
    sort_order: int = 0


class WorkTemplateUpdate(BaseModel):
    """Schema for updating a reusable work template."""
    model_config = ConfigDict(extra="forbid")

    name: Optional[str] = Field(default=None, min_length=1, max_length=255)
    description: Optional[str] = None
    template_type: Optional[TemplateType] = None
    default_title: Optional[str] = Field(default=None, min_length=1, max_length=500)
    default_description: Optional[str] = None
    default_priority: Optional[int] = Field(default=None, ge=1, le=10)
    default_effort_days: Optional[float] = Field(default=None, ge=0.1)
    default_labels: Optional[list[str]] = None
    default_checklist: Optional[list[str]] = None
    default_payload: Optional[dict[str, Any]] = None
    is_active: Optional[bool] = None
    sort_order: Optional[int] = None


class WorkTemplateResponse(BaseModel):
    """Schema for reusable work template responses."""
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    description: Optional[str] = None
    seed_key: Optional[str] = None
    template_type: str
    default_title: Optional[str] = None
    default_description: Optional[str] = None
    default_priority: Optional[int] = None
    default_effort_days: Optional[float] = None
    default_labels: list[str] = Field(default_factory=list)
    default_checklist: list[str] = Field(default_factory=list)
    default_payload: dict[str, Any] = Field(default_factory=dict)
    is_active: bool
    sort_order: int
    created_at: datetime
    updated_at: datetime
